velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -323,6 +323,25 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           const auto& child = children[0];
           auto type = child->type();
           isRepeated = true;
+          if (schemaElement.converted_type == thrift::ConvertedType::LIST &&
+              (child->type()->kind() == TypeKind::MAP ||
```

Comment:
I have the same question. Why did you test `schemaElement.repetition_type == thrift::FieldRepetitionType::REPEATED` instead of
`if (schemaElement.converted_type == thrift::ConvertedType::LIST &&  ((child->type()->kind() == TypeKind::MAP || (child->type()->kind() == TypeKind::ARRAY))`?

Reply:
We cannot check for `child->type()->kind() == TypeKind::ARRAY)` and instead we need to check for `schemaElement.repetition_type == thrift::FieldRepetitionType::REPEATED` because of backward compatibility rule number  mentioned [here](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules) where we have a leaf repeated node and its parent being an optional element of type LIST. 

For example :
```
// List<Integer> (nullable list, non-null elements)
optional group my_list (LIST) {
  repeated int32 element;
}
```
In such cases we create an ARRAY Type for leaf node (  `repeated int32 element` ) since it is repeated [here](https://github.com/facebookincubator/velox/pull/9728/files#diff-64787e76c1b0ad12b5764770a94acd62054896a762ccead8f083a71a060f2f44R522) in the code. So during schema parsing when we parsing the schema element ( `optional group my_list (LIST)` ) , even though it is an element of type LIST and its child is of Type LIST we don't want to create another ARRAY type node here.

Reply:
@hitarth Thanks for explaining. It seems the check should be saying "If either the current schemaElement is thrift::ConvertedType::LIST, and (its repetition_type is REPEATED, or its only child schemaElement is REPEATED), then this node should be a single level or ARRAY". The checking of the child element is MAP should be contained in the condition that the child schemaElement is REPEATED. Is this correct?

Reply:
I tried to check for child schemaElement being REPEATED which could have taken care of both MAP and LIST, it did work for MAP but failing for different LIST case. Hence I had to keep the exclusive child MAP check along with REPEATED check for current element